### PR TITLE
style: Add package.json linter

### PR DESCRIPTION
**Pull request by roninjin10**
Adds a really basic linter for package.json

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -22,3 +22,6 @@ jobs:
 
       - name: Lint
         run: pnpm lint
+
+      - name: Lint package.json
+        run: yarn lint:package-json && git diff --exit-code
```
Comment (roninjin10) [inline .github/workflows/test.yml:27]: should rename this sort-package-json or just use the npx command maybe

File: package.json [hunk 2]
```diff
@@ -1,15 +1,47 @@
...
+    "docs": "pnpm docs:prepare && retype build",
+    "docs:dev": "retype watch .",
+    "docs:prepare": "lerna run docs && pnpm retype:updateversion",
+    "entry:dist": "lerna run prepack",
+    "entry:src": "lerna run postpack",
+    "foundryup": "curl -L https://foundry.paradigm.xyz | bash && bash ~/.foundry/bin/foundryup",
+    "gas-report": "pnpm recursive run gas-report",
+    "link:packages": "lerna exec 'pnpm link --global'",
+    "lint": "pnpm prettier:check && eslint . --ext .ts",
+    "lint:package-json": "npx sort-package-json package.json && npx sort-package-json packages/*/package.json",
```
Comment (roninjin10) [inline package.json:25]: I don't like this name looking at this now

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -22,3 +22,6 @@ jobs:
 
       - name: Lint
         run: pnpm lint
+
+      - name: Lint package.json
+        run: yarn sort-package-json
```
Comment (roninjin10) [inline .github/workflows/test.yml:27]:
```suggestion
        run: pnpm sort-package-json
```

File: packages/services/OPCraftDataSchema.json
Comment (alvrs) [inline packages/services/OPCraftDataSchema.json:1]: Weird that it updated this file and `typedoc.json` too instead of just `package.json`?

**Review by alvrs — DISMISSED** [review]
Very nice! Couple questions, but nothing blocking

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -22,3 +22,6 @@ jobs:
 
       - name: Lint
         run: pnpm lint
+
+      - name: Lint package.json
+        run: pnpm sort-package-json && git diff --exit-code
```
[the remark this answers is not in the document]
Comment (roninjin10) [reply .github/workflows/test.yml:27]: oh nice didn't notice that action I'll use that instead to be consistent

File: packages/services/OPCraftDataSchema.json
Comment (roninjin10) [reply packages/services/OPCraftDataSchema.json:1]: I think this was just user error by me typing the command wrong but I'll revert these files and try again to test that

File: .github/workflows/test.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (holic) [reply .github/workflows/test.yml:27]: ohh could easily swap to this command too, can’t recall why I landed on the git diff approach I am using

**Review by holic — APPROVED** [review]

**Pull request merged**